### PR TITLE
Combine MapLayer and MapCollection search pages

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  label,
+  withPagination,
+  isOverviewPage,
+}) => {
+  // Get the total count for all data types
+  const totalCount = results.reduce((acc, cur) => acc + cur.count, 0)
+  const mapLayerData = results.find(({ type }) => type === MapType.Layer)
+  const mapCollectionData = results.find(({ type }) => type === MapType.Collection)
+
+  return totalCount > 0 ? (
```
Comment (royderks) [inline src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx:39]: The whitespace between the layers and collections seems to be missing here:

Screenshot 2020-04-07 at 09 55 30

File: src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      <>
+        <SearchHeading
+          label={`${label} (${totalCount.toLocaleString(DEFAULT_LOCALE)})`}
+          icon={icon}
+        />
+        <ResultsComponent results={results} />
+        <Spacer />
+        {!withPagination && !isOverviewPage && (
+          // TODO: Select proper filter.
+          <SearchLink
```
Comment (royderks) [inline src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx:109]: For data results we only show the link when the number of results exceeds the "first page", when on the `/data/zoek` page

File: src/app/pages/SearchPage/SearchPageResults.jsx [hunk 3]
```diff
@@ -103,6 +103,11 @@ const SearchPageResults = ({
 }) => {
   const dispatch = useDispatch()
   const allResultsPageActive = currentPage === PAGES.SEARCH
+  const pageConfig = SEARCH_PAGE_CONFIG[currentPage]
+
+  if (!pageConfig) {
+    throw new Error('Unable to find page config')
```
Comment (chrisvanmook) [inline src/app/pages/SearchPage/SearchPageResults.jsx:109]: Will this error be caught somewhere? Otherwise it'll break the whole page..

Comment (jonkoops) [reply src/app/pages/SearchPage/SearchPageResults.jsx:109]: No, this error is not supposed to occur so we would have to find it during development.

File: src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx (hunk 2 above)
Comment (jonkoops) [reply src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx:109]: Ah, yes. I've added this check, also removed the void TODO.

File: src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx (hunk 1 above)
Comment (jonkoops) [reply src/app/pages/SearchPage/MapSearchResults/MapSearchResults.tsx:39]: Fixed

File: src/app/pages/SearchPage/SearchPageResults.jsx (hunk 3 above)
Comment (jonkoops) [reply src/app/pages/SearchPage/SearchPageResults.jsx]: Changed it to not render instead

File: src/app/models/filter.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 export enum FilterType {
-  Checkbox = 'checkbox',
-  Radio = 'radio',
-  Select = 'select',
+  Checkbox = 'CHECKBOX',
```
Comment (royderks) [inline src/app/models/filter.ts:2]: Should this be changed back now the enum in the GraphQL schema couldn't be added?

Comment (jonkoops) [reply src/app/models/filter.ts:2]: Nope, not needed. I have also changed the implementation in the backend.

**Pull request merged**